app/controllers/products_controller.rb

```diff
@@ -2,33 +2,24 @@
 
 class ProductsController < ApplicationController
   before_action :load_product, only: :show
-  before_action :show_slider, only: %i[index show]
-  before_action :load_categories, only: :index
+  before_action :show_slider, only: %i(index show)
 
   def index
-    @pagy, @products = pagy(filter_products, items: Settings.PAGE_9)
-    @product_outstandings = Product.product_outstanding
+    @q = Product.ransack(params[:q])
+    @pagy, @products = pagy(@q.result(distinct: true).order(created_at: :desc).preload(image_attachment: :blob).includes(:comments),
```

Comment:
```suggestion
    @q = Product.ransack(params[:q])
    @pagy, @products = pagy(@q.result(distinct: true).order(created_at: :desc).preload(:comments, image_attachment: :blob),
```
    Tương tự cho phía dưới